src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPath.cs

```diff
@@ -73,23 +77,26 @@ public ODataPathSegment LastSegment
                 return this.segments.LastOrDefault();
             }
         }
-
         /// <summary>
-        /// Get the number of segments in this path.
+        /// Get the List of ODataPathSegments.
         /// </summary>
-        public int Count
+        internal IList<ODataPathSegment> Segments

```

Comment:
Can we have a `ReadOnlyCollection` similar to this https://github.com/OData/WebApi/blob/7a3372ac031968996141de9f77ff0ac93333e642/src/Microsoft.AspNet.OData.Shared/Routing/ODataPath.cs#L94